app/code/Magento/Inventory/Model/SourceItem/Validator/SourceIdValidator.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Inventory\Model\SourceItem\Validator;
+
+use Magento\Framework\Validation\ValidationResult;
+use Magento\Framework\Validation\ValidationResultFactory;
+use Magento\InventoryApi\Api\Data\SourceItemInterface;
+
+/**
+ * Check that source id is valid
+ */
+class SourceIdValidator implements SourceItemValidatorInterface
+{
+    /**
+     * @var ValidationResultFactory
+     */
+    private $validationResultFactory;
+
+    /**
+     * @param ValidationResultFactory $validationResultFactory
+     */
+    public function __construct(ValidationResultFactory $validationResultFactory)
+    {
+        $this->validationResultFactory = $validationResultFactory;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function validate(SourceItemInterface $source): ValidationResult
+    {
+        $errors = [];
+        if (!is_numeric($source->getSourceId())) {
```

Comment:
`is_numeric` would return true for any of the following:
* Signed numbers (but should id be negative?)
* Numbers with decimal part (Should not be used as id?)
* Numbers with exponential part (same)